r/work.R

```diff
@@ -30,29 +14,20 @@ load_data <- function() {
                 print("Current working directory:")
                 print(getwd())
                 print("Experiment folder status:")
-                print(list.files(paste("/data",experiment_id,sep = "/"),all.files=TRUE,full.names=TRUE))
-                f <- readRDS(
-                    paste(
-                        "/data",experiment_id,"r.rds",
-                        sep = "/"
-                    )
-                )
-                loaded <- T
+                print(list.files(dirname(fpath), all.files=TRUE, full.names=TRUE))
+                f <- readRDS(fpath)
+                loaded <- TRUE
                 length <- dim(f)
-                message(
-                    paste(
-                        "Data successfully loaded, dimensions",
-                        length[1], "x", length[2]
-                    )
-                )
+                message("Data successfully loaded, dimensions",
+                        length[1], "x", length[2])
 
-                f
+                return(f)
             },
             warning = function(w) {
-                message(paste("file could not be loaded: ", w))
+                message("file could not be loaded: ", w)
             },
             error = function(e) {
-                message(paste("file could not be loaded: ", e))
+                message("file could not be loaded: ", e)
             }
         )
         Sys.sleep(1)
```

Comment:
Why do we have this sleep here?

Reply:
I'm not sure. I suspect it's in case the data has not yet been downloaded fully then the function fails, waits, and tries to load it again.

Reply:
Yes, that's correct. The loop is to poll until the file exists.